{lib}[foss/2020a] TensorFlow v2.3.1 w/ Python 3.8.2

terjekv opened the pull request:
(created using `eb --new-pr`)

Requires ~https://github.com/easybuilders/easybuild-easyconfigs/pull/11543~

boegel commented on easybuild/easyconfigs/h/hwloc/hwloc-1.11.12-GCCcore-9.3.0.eb:
```diff
@@ -0,0 +1,48 @@
+easyblock = 'ConfigureMake'
+
+name = 'hwloc'
+version = '1.11.12'
```
@terjekv This one means trouble, since we already have an `hwloc` as a part of `foss/2020a` (as a dep for OpenMPI).

Is `hwloc` 2.2.0 (via ` hwloc-2.2.0-GCCcore-9.3.0.eb`) not an option?

terjekv replied:
Oh, I thought we treated `hwloc-1*` and `hwlock-2` as different projects. I can test `hwloc-2.2.0-GCCcore-9.3.0.eb`

Flamefire replied:
That is even what TF expects :)

terjekv replied:
Tested with `hwloc-2.2.0-GCCcore-9.3.0.eb`. Seems ok. 👍

boegel commented on easybuild/easyconfigs/j/JsonCpp/JsonCpp-1.9.3-GCCcore-9.3.0.eb:
```diff
@@ -0,0 +1,29 @@
+easyblock = "CMakeNinja"
+
+name = 'JsonCpp'
+version = '1.9.3'
```
This should be updated to `1.9.4`?

terjekv replied:
That sounds reasonable.

terjekv replied:
Upgraded to `1.9.4`. Seems ok.

boegel commented on easybuild/easyconfigs/p/protobuf-python/protobuf-python-3.10.0-foss-2020a-Python-3.8.2.eb:
```diff
@@ -0,0 +1,34 @@
+easyblock = 'PythonPackage'
+
+name = 'protobuf-python'
+version = '3.10.0'
```
I wonder how strictly we should stick to this version, also taking into account future TensorFlow version that we'll likely install with `foss/2020a`.

@Flamefire Can we consider a more recent `protobuf-python` here, like latest stable?

Flamefire replied:
Current TF master is still using 3.9.2: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/workspace.bzl#L615

I'd say try it ;)

terjekv replied:
It would seem we already have a `protobuf/3.10.0` for `foss/2020a`?

terjekv replied:
Gave it an update anyway, it seems to work as expected!

boegel commented on easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.3.1-foss-2020a-Python-3.8.2.eb:
```diff
@@ -0,0 +1,165 @@
+# Dependencies created and updated using findPythonDeps.sh:
+# https://gist.github.com/Flamefire/49426e502cd8983757bd01a08a10ae0d
+exts_list = [
+    ('Markdown', '3.2.2', {
```
@terjekv These should be revisited, some may already included with `Python`, some should probably be updated?

Flamefire replied:
What I did: `eb --dump-env` on this EC, then source it and run the linked script. That should give you a nice list of packages and versions to use.

smoors commented:
@terjekv you've added hwloc-1.11.12, but you're not actually using it in the TF easyconfig, is that on purpose?

terjekv commented:
> @terjekv you've added hwloc-1.11.12, but you're not actually using it in the TF easyconfig, is that on purpose?

Bah, thanks. I moved the PR to use hwloc-2.2.0 and forgot to remove the old one. Fixed now. Thanks again!

smoors commented:
Test report by @smoors
**FAILED**
Build succeeded for 8 out of 9 (7 easyconfigs in total)
node316.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/5c62e0788eaef18cd669e164123f4a41 for a full test report.

terjekv commented:
> Test report by @smoors
> **FAILED**
> Build succeeded for 8 out of 9 (7 easyconfigs in total)
> node316.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
> See https://gist.github.com/5c62e0788eaef18cd669e164123f4a41 for a full test report.

Nice catch.

```
== 2020-10-21 22:02:54,945 modules.py:809 DEBUG Output of module command '/usr/share/lmod/lmod/libexec/lmod python load protobuf-python/3.13.0-foss-2020a-Python-3.8.2': stdout: _mlstatus = False
; stderr: Lmod has detected the following error: A different version of the 'protobuf'
module is already loaded (see output of 'ml').
You should load another 'protobuf-python' module that is compatible with the
currently loaded version of 'protobuf'.
Use 'ml spider protobuf-python' to get an overview of the available versions.


If you don't understand the warning or error, contact the helpdesk at
hpc@[...] 
While processing the following module(s):
    Module fullname                                 Module Filename
    ---------------                                 ---------------
    protobuf-python/3.13.0-foss-2020a-Python-3.8.2  /tmp/vsc10009/ebinstall/11546/modules/all/protobuf-python/3.13.0-foss-2020a-Python-3.8.2.lua
```
Should be resolved in https://github.com/easybuilders/easybuild-easyconfigs/pull/11546/commits/8d8dbd615e2bc20610c83589d039fd8ef1e584e1.

smoors commented:
Test report by @smoors
**SUCCESS**
Build succeeded for 9 out of 9 (7 easyconfigs in total)
node379.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/29d7b1b37bda1b46722b1d00242cb385 for a full test report.

smoors reviewed (dismissed):
lgtm

boegel commented:
@boegelbot please test @ generoso

boegel commented:
Test report by @boegel
**FAILED**
Build succeeded for 6 out of 7 (7 easyconfigs in total)
node3401.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/497b3d5cecd3f1c8d47bf8838108353e for a full test report.

boegel commented:
@boegelbot please test @ generoso

terjekv commented:
Test report by @terjekv
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
arm2 - Linux ubuntu 18.04, AArch64, UNKNOWN, Python 3.6.9
See https://gist.github.com/9bc903a2fd3009f7a6e122dcaf0d364b for a full test report.

branfosj commented:
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (7 easyconfigs in total)
bear-pg0305u03a.bear.cluster - Linux RHEL 7.6, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/0cd2496e94ab194e7178f8d02283d0bc for a full test report.

boegel commented:
@branfosj Hmm...

```
ImportError: Traceback (most recent call last):
  File "/dev/shm/build-branfosj-admin/branfosj-admin-up/TensorFlow/2.3.1/foss-2020a-Python-3.8.2/tmpja4zujxf-bazel-tf/output_base/execroot/org_tensorflow/bazel-out/ppc-opt/bin/tensorflow/python/keras/api/create_tensorflow.python_api_keras_python_api_gen_compat_v2.runfiles/org_tensorflow/tensorflow/python/pywrap_tensorflow.py", line 64, in <module>
    from tensorflow.python._pywrap_tensorflow_internal import *
ImportError: /dev/shm/build-branfosj-admin/branfosj-admin-up/TensorFlow/2.3.1/foss-2020a-Python-3.8.2/tmpja4zujxf-bazel-tf/output_base/execroot/org_tensorflow/bazel-out/ppc-opt/bin/tensorflow/python/keras/api/create_tensorflow.python_api_keras_python_api_gen_compat_v2.runfiles/org_tensorflow/tensorflow/python/_pywrap_tensorflow_internal.so: undefined symbol: _ZTVN6icu_669ErrorCodeE


Failed to load the native TensorFlow runtime.
```

=> `_pywrap_tensorflow_internal.so: undefined symbol: _ZTVN6icu_669ErrorCodeE`

@Flamefire Any ideas?

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
node3401.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/7487cc7d0525cd386ffead6249728030 for a full test report.

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
node3107.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/cce41122794e01e6155fefd78097083d for a full test report.

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
node2426.golett.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/921c3d311ed4b215b18a2cb624389b66 for a full test report.

branfosj commented:
Test report by @branfosj
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
bear-pg0306u19a.bear.cluster - Linux RHEL 8.2, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/046f4c8bc5eeaed54e9f7cfa6c534e6c for a full test report.

ocaisa commented:
@branfosj That last report looks like it's coming from the same system as the previous one that showed the error, do you know what fixed it?

branfosj commented:
@ocaisa The reports are from different systems. The failed build (bear-pg0305u03a) is on RHEL 7 - with the dependencies built over the last few months. The successful build (bear-pg0306u19a) is on RHEL 8 - with all the dependencies built today. I'm currently trying selective rebuild of the dependencies to see if I can get it to build.

ocaisa commented:
Ok, @surak is seeing a similar error on our systems for centos7 I think

branfosj commented:
Ok. In order I've:

1. rebuilt `pybind11-2.4.3-GCCcore-9.3.0-Python-3.8.2.eb`
1. switched to a direct login (i.e. escaped the cgroup)
1. rebuilt `ICU-66.1-GCCcore-9.3.0.eb`

With those TensorFlow builds for me on the system where it was failing before. I've set off a test report.

I cannot go back and check, but it may just need ICU rebuilding. My guess for rebuilding ICU was hoping that the `icu` in the missing symbol `_ZTVN6icu_669ErrorCodeE` was useful information.

branfosj commented:
Test report by @branfosj
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
bear-pg0305u03a.bear.cluster - Linux RHEL 7.6, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/ccb832d1244ee87ecb7cef51269e8983 for a full test report.

smoors reviewed (approved):
lgtm

smoors commented:
Going in, thanks @terjekv!

branfosj commented:
> I cannot go back and check, but it may just need ICU rebuilding. My guess for rebuilding ICU was hoping that the `icu` in the missing symbol `_ZTVN6icu_669ErrorCodeE` was useful information.

This was fixed by https://github.com/easybuilders/easybuild-framework/pull/3401

casparvl commented:
Just to add for future reference: I hit exactly the same `missing symbol _ZTVN6icu_669ErrorCodeE` that @branfosj hit. That ICU was built with `EasyBuild 4.2.1`. Rebuilding it with `EasyBuild 4.3.1` was sufficient to resolve this error, as @branfosj also anticipated, i.e. his steps 1 and 2 listed in his post above were not needed to resolve it.
